serviceerror/convert.go

```diff
@@ -87,58 +87,52 @@ func FromStatus(st *status.Status) error {
 	}
 
 	errDetails := extractErrorDetails(st)
-
-	// If there was an error during details extraction, it will go to errDetails.
-	if err, ok := errDetails.(error); ok {
```

Comment:
Lets add a unit test for this. Here's how I made an error of unknown type:

```go
err := status.ErrorProto(&statuspb.Status{
	Code:    int32(codes.NotFound),
	Message: "some error!",
	Details: []*anypb.Any{{TypeUrl: "type.googleapis.com/some.unknown.Type"}},
})
```

Reply:
Added.